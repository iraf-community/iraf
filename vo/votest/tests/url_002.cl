#
#  Rudimentary URL handling: images

# Set the test description string.
votest.descr = "Rudimentary URL handling: images"

# Convert the data$logical to a local path.
s1 = data_url // "/sif.fits"
s2 = "file://" // data_path // "/sif.fits"
s3 = "file:///localhost" // data_path // "/sif.fits"

# Execute the test commands.
print ("\nLogical Path:  data$sif.fits")                # logical path
imstat ("data$sif.fits", field="mean,min,max")

print ("\nHTTP URI:  " // s1)                           # remote http URI
imstat (s1, field="mean,min,max")

print ("\nFile URI:  " // s2)                           # file URI
imstat (s2, field="mean,min,max")

print ("\nFile URI:  " // s3)                           # file URI
imstat (s3, field="mean,min,max")

print ("\nHTTP Redirection")				# HTTP Redirection
imstat ("http://iraf.noao.edu/votest/redir_img.php", field="mean,min,max")

print ("\nHTTP CGI query")				# HTTP CGI query
imstat http://iraf.noao.edu/scripts/voget?t=fits&f=sif.fits field="mean,min,max"
imstat ("http://iraf.noao.edu/scripts/voget?t=fits&f=sif.fits", 
		field="mean,min,max")

s1 = "http://iraf.noao.edu/scripts/voget?t=fits&f=sif.fits"
imstat (s1, field="mean,min,max")

