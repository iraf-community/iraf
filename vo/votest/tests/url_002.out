
Logical Path:  data$sif.fits
#      MEAN       MIN       MAX
      108.3       -1.    19936.

HTTP URI:  http://iraf.noao.edu/votest//sif.fits
#      MEAN       MIN       MAX
      108.3       -1.    19936.

File URI:  file:///iraf/iraf/vo/votest/data//sif.fits
#      MEAN       MIN       MAX
      108.3       -1.    19936.

File URI:  file:///localhost/iraf/iraf/vo/votest/data//sif.fits
#      MEAN       MIN       MAX
      108.3       -1.    19936.

HTTP Redirection
#      MEAN       MIN       MAX
      108.3       -1.    19936.

HTTP CGI query
#      MEAN       MIN       MAX
      108.3       -1.    19936.
#      MEAN       MIN       MAX
      108.3       -1.    19936.
#      MEAN       MIN       MAX
      108.3       -1.    19936.
